### ISSUE TITLE
Investigate Corb 2.4.0 performance single corb instance high thread count vs multiple instances with split thread counts

### ISSUE THREAD

**Issue by bbandlamudi** [marklogic-community/corb2#81]
There has been a concern if corb is becoming a bottleneck at high thread counts and need to analyze the performance impact of single instance with high thread count vs multiple corb instances with split thread counts against the same marklogic instance/cluster.

**Comment by hansenmc** [comment]
Any indications of whether there is a difference in the execution times of the process module, or if there are delays between the next execution of the process module?

Has this been noted for all types of jobs, or just for jobs that export (write) data on the client? 
Is the discrepancy the same for jobs that just transform data and don't return/write files on the client?

If there are delays between the threads next module execution, then likely candidates dragging it down would be the synchronized file writes for `ExportBatchToFileTask`, maxing out client disk IO, and/or the coordination of queue items with the `DISK-QUEUE`.

I'm sure that you have already considered most of this, but just throwing things out for discussion and consideration: 
- If using `ExportBatchToFileTask`, do we see the same discrepancy when run with `ExportToFileTask`?
- maybe it's just disk IO and spreading over different machines allows greater write rates?
    - Is there is a difference with multiple CORB jobs running on the same client vs. running them each on separate machines, or writing to a shared filesystem.
- Try eliminating the DISK-QUEUE for pure memory queue to see if it evens out?

**Comment by bbandlamudi** [comment]
This is one of the things that was being investigated (independently) at our project to account some of the discrepancies in performance testing results and corb being a potential bottleneck is raised as a concern. 

I realized that we never did this kind of testing in Corb and I thought it is good to have an answer to this question. Also the current analysis does not need to involve data export, but it is not a bad idea to throw it into the mix and see what the impact is. If corb is in fact a bottleneck, we should think of ways to mitigate it.

**Comment by bbandlamudi** [comment]
I did a bit of local testing and found that increased thread count isn't throttling corb. I tried 1 instance x 48 threads, 2 instances x 24 threads and 4 instances x 12 threads (all on a local box). The total throughput (tps) out of all the running instances was pretty much same in all three cases. 

Also, simple CSV file export is decreasing the throughput (tps) by about 10% compared to the same job that was run with out a CSV export. I believe a little bit of overhead due to file IO is expected and acceptable.
